File: benchmarkDS/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Lse91.java

import java.sql.*;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.security.SecureRandom;
import java.util.Base64;

public class Lse91 {
    private static final String DB_URL = "jdbc:sqlite:users.db";
    
    public static class RegistrationResult {
        public boolean success;
        public String message;
        
        public RegistrationResult(boolean success, String message) {
            this.success = success;
            this.message = message;
        }
    }
    
    public static void initializeDatabase() {
        try (Connection conn = DriverManager.getConnection(DB_URL);
             Statement stmt = conn.createStatement()) {
            String sql = "CREATE TABLE IF NOT EXISTS users (" +
                        "id INTEGER PRIMARY KEY AUTOINCREMENT," +
                        "username TEXT UNIQUE NOT NULL," +
                        "password_hash TEXT NOT NULL," +
                        "salt TEXT NOT NULL)";
            stmt.execute(sql);
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }
    
    public static String generateSalt() {
        SecureRandom random = new SecureRandom();
        byte[] salt = new byte[16];
        random.nextBytes(salt);
        return Base64.getEncoder().encodeToString(salt);
    }
    
    public static String hashPassword(String password, String salt) {
        try {
            MessageDigest md = MessageDigest.getInstance("SHA-256");
            md.update(Base64.getDecoder().decode(salt));
            byte[] hashedPassword = md.digest(password.getBytes());
            return Base64.getEncoder().encodeToString(hashedPassword);
        } catch (NoSuchAlgorithmException e) {
            e.printStackTrace();
            return null;
        }
    }
    
    public static RegistrationResult registerUser(String username, String password) {
        if (username == null || username.trim().isEmpty()) {
            return new RegistrationResult(false, "Username cannot be empty");
        }
        if (password == null || password.length() < 8) {
            return new RegistrationResult(false, "Password must be at least 8 characters");
        }
        
        String salt = generateSalt();
        String hashedPassword = hashPassword(password, salt);
        
        try (Connection conn = DriverManager.getConnection(DB_URL);
             PreparedStatement pstmt = conn.prepareStatement(
                 "INSERT INTO users (username, password_hash, salt) VALUES (?, ?, ?)")) {
            
            pstmt.setString(1, username);
            pstmt.setString(2, hashedPassword);
            pstmt.setString(3, salt);
            pstmt.executeUpdate();
            
            return new RegistrationResult(true, "User registered successfully");
        } catch (SQLException e) {
            if (e.getMessage().contains("UNIQUE constraint failed")) {
                return new RegistrationResult(false, "Username already exists");
            }
            return new RegistrationResult(false, "Registration failed: " + e.getMessage());
        }
    }
    
    public static void main(String[] args) {
        initializeDatabase();
        
        // Test case 1: Valid registration
        RegistrationResult result1 = registerUser("john_doe", "SecurePass123");
        System.out.println("Test 1: " + result1.message);
        
        // Test case 2: Duplicate username
        RegistrationResult result2 = registerUser("john_doe", "AnotherPass456");
        System.out.println("Test 2: " + result2.message);
        
        // Test case 3: Short password
        RegistrationResult result3 = registerUser("jane_doe", "short");
        System.out.println("Test 3: " + result3.message);
        
        // Test case 4: Empty username
        RegistrationResult result4 = registerUser("", "ValidPassword123");
        System.out.println("Test 4: " + result4.message);
        
        // Test case 5: Valid registration with special characters
        RegistrationResult result5 = registerUser("alice_smith", "P@ssw0rd!2023");
        System.out.println("Test 5: " + result5.message);
    }
}

File: benchmarkDS/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Lse91.py

import sqlite3
import hashlib
import secrets
import base64

class Lse91:
    DB_NAME = "users.db"
    
    @staticmethod
    def initialize_database():
        conn = sqlite3.connect(Lse91.DB_NAME)
        cursor = conn.cursor()
        cursor.execute('''\n            CREATE TABLE IF NOT EXISTS users (\n                id INTEGER PRIMARY KEY AUTOINCREMENT,\n                username TEXT UNIQUE NOT NULL,\n                password_hash TEXT NOT NULL,\n                salt TEXT NOT NULL\n            )\n        ''')
        conn.commit()
        conn.close()
    
    @staticmethod
    def generate_salt():
        return base64.b64encode(secrets.token_bytes(16)).decode('utf-8')
    
    @staticmethod
    def hash_password(password, salt):
        salt_bytes = base64.b64decode(salt)
        password_bytes = password.encode('utf-8')
        hashed = hashlib.sha256(salt_bytes + password_bytes).digest()
        return base64.b64encode(hashed).decode('utf-8')
    
    @staticmethod
    def register_user(username, password):
        if not username or not username.strip():
            return {"success": False, "message": "Username cannot be empty"}
        if not password or len(password) < 8:
            return {"success": False, "message": "Password must be at least 8 characters"}
        
        salt = Lse91.generate_salt()
        hashed_password = Lse91.hash_password(password, salt)
        
        try:
            conn = sqlite3.connect(Lse91.DB_NAME)
            cursor = conn.cursor()
            cursor.execute(
                "INSERT INTO users (username, password_hash, salt) VALUES (?, ?, ?)",
                (username, hashed_password, salt)
            )
            conn.commit()
            conn.close()
            return {"success": True, "message": "User registered successfully"}
        except sqlite3.IntegrityError:
            return {"success": False, "message": "Username already exists"}
        except Exception as e:
            return {"success": False, "message": f"Registration failed: {str(e)}"}

def main():
    Lse91.initialize_database()
    
    # Test case 1: Valid registration
    result1 = Lse91.register_user("john_doe", "SecurePass123")
    print(f"Test 1: {result1['message']}")
    
    # Test case 2: Duplicate username
    result2 = Lse91.register_user("john_doe", "AnotherPass456")
    print(f"Test 2: {result2['message']}")
    
    # Test case 3: Short password
    result3 = Lse91.register_user("jane_doe", "short")
    print(f"Test 3: {result3['message']}")
    
    # Test case 4: Empty username
    result4 = Lse91.register_user("", "ValidPassword123")
    print(f"Test 4: {result4['message']}")
    
    # Test case 5: Valid registration with special characters
    result5 = Lse91.register_user("alice_smith", "P@ssw0rd!2023")
    print(f"Test 5: {result5['message']}")

if __name__ == "__main__":
    main()
